.github/workflows/main.yml

```diff
@@ -8,16 +8,53 @@ on:
   pull_request:
     branches: [ master ]
 
-# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+
+env:
+  CI: true
+
+
+# Job progression.  We make sure that the base image [oss] builds and passes tests before kicking off the other builds
+
+    #                                ┌──────────────────┐        ┌────────────────┐      ┌────────────────┐
+    # ┌─────────┐     ┌─────────┬────► Build Latest NJS ├────────►Test Latest NJS ├─────►│Push Latest NJS │
+    # │Build OSS├────►│Test OSS │    └──────────────────┘        └────────────────┘      └────────────────┘
+    # └─────────┘     └──┬──────┤
+    #                    │      │    ┌──────────────────┐       ┌──────────────────┐     ┌─────────────────┐
+    #                    │      └────►Build Unprivileged├───────►Test Unprivileged ├────►│Push Unprivileged│
+    #                    │           └──────────────────┘       └──────────────────┘     ├────────┬────────┘
+    #                    │                                                               ├────────┤
+    #                    └──────────────────────────────────────────────────────────────►│Push OSS│
+    #                                                                                    └────────┘
+
+# As a last step, if we are on the main/master branch, multi-architecture images will be built and pushed to github packages
+# and docker hub
+
 jobs:
-  build:
+  build-oss-for-test:
     runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+      - name: Build and export
+        uses: docker/build-push-action@v5
+        with:
+          file: Dockerfile.oss
+          context: .
+          tags: nginx-s3-gateway , nginx-s3-gateway:oss
+          outputs: type=docker,dest=/tmp/oss.tar
+      - name: Upload artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: oss
+          path: /tmp/oss.tar
```

Comment:
**annotation**: The other two images use this as a base.  Therefore we export the built image for use in:
* The tests for this image
* The build process for the `latest-njs` and `unprivileged` images